src/server/query.js

```diff
@@ -1,62 +1,20 @@
-import R from 'ramda';
 import schemas from '../schema';
+import {
+  createGroupAggregation,
+  getQueryKeys,
+  hasMeasurePrefix,
+} from './utils';
 
-export const decisionQuery = (obj, args, context, resolveInfo) => {
-  const prepend = R.ifElse(
-    R.test(/^measure/),
-    R.concat('$measures.'),
-    R.concat('$')
-  );
-
-  const aggregationOperator = R.cond([
-    [R.equals('count'), R.always({ $sum: 1 })],
-    [R.equals('month'), R.always({ $first: { $month: prepend('date') } })],
-    [R.T, key => ({ $first: prepend(key) })],
-  ]);
-
-  const formId = R.cond([
-    [R.equals('count'), R.always('')],
-    [
-      R.equals('month'),
-      R.always({ $substr: [{ $month: prepend('date') }, 0, -1] }),
-    ],
-    [R.T, key => ({ $substr: [prepend(key), 0, -1] })],
-  ]);
-
-  const queryKeys = resolveInfo.operation.selectionSet.selections[
-    0
-  ].selectionSet.selections.map(selection => selection.name.value);
-
-  const shouldUnwind = R.any(k => k.startsWith('measure'), queryKeys);
+const sort = { $sort: { _id: -1 } };
+const unwindOperator = { $unwind: '$measures' };
 
-  const id = {
-    $concat: R.intersperse('|', queryKeys.map(formId)),
-  };
-  console.log(R.intersperse('|', queryKeys.map(k => `$${k}`)));
-  console.log(id);
-
-  const groupObj = {
-    $group: queryKeys.reduce(
-      (o, key) =>
-        Object.assign({}, o, {
-          [key]: aggregationOperator(key),
-        }),
-      { _id: id }
-    ),
-  };
-
-  const sort = {
-    $sort: { _id: -1 },
-  };
-
-  console.log(groupObj);
-
-  const unwindOperator = { $unwind: '$measures' };
-  const pipeline = shouldUnwind
-    ? [unwindOperator, groupObj, sort]
-    : [groupObj, sort];
+export const decisionQuery = (obj, args, context, resolveInfo) => {
+  const queryKeys = getQueryKeys(resolveInfo);
+  const group = createGroupAggregation(queryKeys);
 
-  const resultPromise = schemas.Decision.aggregate(pipeline).exec();
+  const pipeline = hasMeasurePrefix(queryKeys)
+    ? [unwindOperator, group, sort]
+    : [group, sort];
 
-  return resultPromise;
+  return schemas.Decision.aggregate(pipeline).exec();
```

Comment:
Yeah, I would not go that far into the functional world here. I like the style that's going to be more maintainable by the future developers.